doc/api/process.md

```diff
@@ -147,20 +147,32 @@ most convenient for scripts).
 ### Event: 'uncaughtException'
 <!-- YAML
 added: v0.1.18
+changes:
+  - version: REPLACEME
+    pr-url: https://github.com/nodejs/node/pull/20097
+    description: Garbage collected unhandled rejections will now also trigger an
+                 uncaughtException. The `errorOrigin` argument was added to
+                 distinguish those.
 -->
 
+* `err` {Error} The uncaught exception.
+* `errorOrigin` {string} Indicates if the exception originates from an
+  unhandled rejection or from synchronous errors. Can either be `'FROM_PROMISE'`
+  or `'FROM_ERROR'`.
+
 The `'uncaughtException'` event is emitted when an uncaught JavaScript
 exception bubbles all the way back to the event loop. By default, Node.js
 handles such exceptions by printing the stack trace to `stderr` and exiting.
 Adding a handler for the `'uncaughtException'` event overrides this default
 behavior.
 
-The listener function is called with the `Error` object passed as the only
-argument.
-
 ```js
-process.on('uncaughtException', (err) => {
-  fs.writeSync(1, `Caught exception: ${err}\n`);
+process.on('uncaughtException', (err, errorOrigin) => {
```

Comment:
Nit: this looks more like our node.js style callback now :D

Reply:
The `unhandledRejection` hook looks similar as well. But I guess you mean this because of the `err` as first argument? I would like to have a different way dealing with this but after playing with a alternative, I decided to still stick to this.